eslint-plugin-prettier.js

```diff
@@ -0,0 +1,306 @@
+
+// Lazily-loaded Prettier.
+let prettier;
+
+// ------------------------------------------------------------------------------
+//  Helpers
+// ------------------------------------------------------------------------------
+
+/**
+ * Gets the location of a given index in the source code for a given context.
+ * @param {RuleContext} context - The ESLint rule context.
+ * @param {number} index - An index in the source code.
+ * @returns {Object} An object containing numeric `line` and `column` keys.
+ */
+function getLocFromIndex(context, index) {
+  // If `sourceCode.getLocFromIndex` is available from ESLint, use it.
+  // Otherwise, use the private version from eslint/lib/ast-utils.
+  // `sourceCode.getLocFromIndex` was added in ESLint 3.16.0.
+  const sourceCode = context.getSourceCode();
+  if (typeof sourceCode.getLocFromIndex === 'function') {
+    return sourceCode.getLocFromIndex(index);
+  }
+  const astUtils = require('eslint/lib/ast-utils');
+  return astUtils.getLocationFromRangeIndex(sourceCode, index);
+}
+
+/**
+ * Converts invisible characters to a commonly recognizable visible form.
+ * @param {string} str - The string with invisibles to convert.
+ * @returns {string} The converted string.
+ */
+function showInvisibles(str) {
+  let ret = '';
+  for (let i = 0; i < str.length; i++) {
+    switch (str[i]) {
+      case ' ':
+        ret += '\u00B7'; // Middle Dot
+        break;
+      case '\n':
+        ret += '\u23ce'; // Return Symbol
+        break;
+      case '\t':
+        ret += '\u21b9'; // Left Arrow To Bar Over Right Arrow To Bar
+        break;
+      default:
+        ret += str[i];
+        break;
+    }
+  }
+  return ret;
+}
+
+// ------------------------------------------------------------------------------
+//  Rule Definition
+// ------------------------------------------------------------------------------
+
+/**
+ * Reports issues where the context's source code differs from the Prettier
+ formatted version.
+ * @param {RuleContext} context - The ESLint rule context.
+ * @param {string} prettierSource - The Prettier formatted source.
+ * @returns {void}
+ */
+function reportDifferences(context, prettierSource) {
+  // fast-diff returns the differences between two texts as a series of
+  // INSERT, DELETE or EQUAL operations. The results occur only in these
+  // sequences:
+  //           /-> INSERT -> EQUAL
+  //    EQUAL |           /-> EQUAL
+  //           \-> DELETE |
+  //                      \-> INSERT -> EQUAL
+  // Instead of reporting issues at each INSERT or DELETE, certain sequences
+  // are batched together and are reported as a friendlier "replace" operation:
+  // - A DELETE immediately followed by an INSERT.
+  // - Any number of INSERTs and DELETEs where the joining EQUAL of one's end
+  // and another's beginning does not have line endings (i.e. issues that occur
+  // on contiguous lines).
+
+  const source = context.getSource();
```

Comment:
I think `context.getSource()` is deprecated -- there doesn't seem to be any documentation for it, and it's in the list of deprecated passthroughs [here](https://github.com/eslint/eslint/blob/0ef09ea071f7df0cfb3f76ddea9ad746cb619912/lib/rule-context.js#L24-L41). Can you replace it with `context.getSourceCode().text` instead?